### DISCUSSION TITLE
Is it possible to do only sort without formatting the import?

### DISCUSSION THREAD

**Discussion by ecneil** [astral-sh/ruff#8734] [Q&A]
I am considering switching from flake8.
(since pydocstyle has been deprecated)

So far, I have been using `flake8-import-order` to sort imports.
I tried ruff's isort, and it seems that the lint rule of isort is affected by formatter. Even if the sort order is correct, it tries to split a long import into individual lines, and an error occurs.

I don't like this behavior because I don't like too long vertical lines because they are not legible to me.

Is there a way to ignore format and only lint about the order of imports?

**Comment by charliermarsh** [comment]
Unfortunately, if I understand the question correctly, I don't think we support what you're looking for, which is: _not_ wrapping imports that exceed a certain line length.

**Reply by ecneil** [reply]
Sorry I am not explaining it well.

I was trying and it was formatted as follows, which is why I asked the question.

Original
```
from package import ManyImport1, ManyImport2, ManyImport3, ManyImport4, ManyImport5, ManyImport6, ManyImport7
```

I want
```
from package import (
    ManyImport1, ManyImport2, ManyImport3, ManyImport4, ManyImport5,
    ManyImport6, ManyImport7)
```

ruff suggest
```
from package import (
    ManyImport1,
    ManyImport2,
    ManyImport3,
    ManyImport4,
    ManyImport5,
    ManyImport6,
    ManyImport7
)
```

**Reply by MichaReiser** [reply]
Could it be that you're looking for the [multi_line_output](https://github.com/astral-sh/ruff/issues/2600) which ruff doesn't support (yet?). If not, could you share your existing isort configuration (assuming you used isort before)

**Reply by peter-gerlagh-kyos** [reply]
I have a very similar desire. In my `.pyi` files I want ruff to ignore the line-length. This mostly works, but not for the imports. So I want:
```
from xxxxxxxx_asyncio_client.models.delivery_area import DeliveryArea  # type: ignore[import-untyped]
```
but I get
```
from xxxxxxxx_asyncio_client.models.delivery_area import (
    DeliveryArea,  # type: ignore[import-untyped]
)
```

I also tried setting the line length to a large value for `.pyi` files. But that doesn't seem to work either.

For the non-import lines,
```
[lint.per-file-ignores]
"**/*.pyi" = [
  "E501",  # ignore line length for type stub files
]
```
combined with
```
fmt: off
```
does the trick.
